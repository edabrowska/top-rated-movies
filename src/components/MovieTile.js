import React from 'react'

const MovieTile = ({ title, releaseDate, averageVote, onClick }) => {
  return (
    <li className='movie-tile'>
      <div className='movie-tile__wrapper-left'>
        <h3 className='movie-tile__title'>{title}</h3>
        <p>{releaseDate}</p>
      </div>
      <div className='movie-tile__wrapper-right'>
        <p className='movie-tile__rating'>{averageVote}</p>
        <button className='movie-tile__button' onClick={onClick}>INFO</button>
      </div>
    </li>
  )
}

export default MovieTile
