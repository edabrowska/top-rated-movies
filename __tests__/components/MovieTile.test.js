import React from 'react'
import renderer from 'react-test-renderer'

import MovieTile from '../../src/components/MovieTile'

describe('MovieTile', () => {
  it('Snapshot', () => {
    const component = renderer.create(
      <MovieTile
        title="Ad Astra"
        releaseDate="2019-09-17"
        averageVote={9}
        onClick={() => 'asd'}
      />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
