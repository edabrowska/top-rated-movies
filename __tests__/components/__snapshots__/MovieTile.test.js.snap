// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTile Snapshot 1`] = `
<li
  className="movie-tile"
>
  <div
    className="movie-tile__wrapper-left"
  >
    <h3
      className="movie-tile__title"
    >
      Ad Astra
    </h3>
    <p>
      2019-09-17
    </p>
  </div>
  <div
    className="movie-tile__wrapper-right"
  >
    <p
      className="movie-tile__rating"
    >
      9
    </p>
    <button
      className="movie-tile__button"
      onClick={[Function]}
    >
      INFO
    </button>
  </div>
</li>
`;
